docs/kubernetes.md

```diff
@@ -167,9 +167,150 @@ Max Accuracy: 100.00%
 
 ## TPUs
 
-For TPUs, the same steps for `torchrun` apply: create one job instance per host
-(`--nnodes`) and one process per chip (`--nprocs_per_node`). For more
+Training on TPU is similar to training on GPU in GKE, the same steps for `torchrun` apply. For more
 information about TPU GKE clusters, see [GKE's official
 docs](https://cloud.google.com/kubernetes-engine/docs/how-to/tpus).
 
-TODO: TPU example
+The example below use two ct5lp-hightpu-4t VMs, with 4 v5e TPU each to construct a 2x4 topology nodepool.
+You can adjust the values accordingly to match the training requirement.
+
+Create a new file `tpu_test.yaml` with the following:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: headless-svc
+  namespace: default
+spec:
+  clusterIP: None
+  selector:
+    job-name: torch-xla-tpu-2x4
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: torch-xla-tpu-2x4
+  labels:
+spec:
+  parallelism: 2 # num of nodes
+  completions: 2 # num of nodes
+  backoffLimit: 0 # default, no retries
+  completionMode: Indexed
+  template:
+    spec:  # pod-spec:
+      serviceAccountName: default
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions: # need to be specified to get tpu resources
+              - key: cloud.google.com/gke-tpu-accelerator
+                operator: "In"
+                values:
+                - "tpu-v5-lite-podslice"
+              - key: cloud.google.com/gke-tpu-topology
+                operator: "In"
+                values:
+                - "2x4"  # 2 nodes of 4 tpu's
+      tolerations:
+      - key: "google.com/tpu"
+        operator: "Equal"
+        value: "present"
+        effect: "NoSchedule"
+      restartPolicy: Never # look in https://kubernetes.io/docs/concepts/workloads/controllers/job/
+      subdomain: headless-svc
+      volumes:
+        # Increase size of tmpfs /dev/shm to avoid OOM.
+      - name: shm
+        emptyDir:
+          medium: Memory
+      containers:
+      - name: training
+        image: us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/xla:nightly_3.10_tpuvm
+        command:
+        - bash
+        - -cxue
+        - |
+          apt-get update -y
```

Comment:
The additional pip installs and symlinking should no longer be necessary with any recent release

Reply:
removed